File: 0036-valid-sudoku/0036-valid-sudoku.go
func isValidSudoku(board [][]byte) bool {
    seen := make(map[string]bool)
    
    for i := 0; i < 9; i++ {
        for j := 0; j < 9; j++ {
            number := board[i][j]
            if number != '.' {
                row := string(number) + " seen in row " + string(i)
                col := string(number) + " seen in col " + string(j)
                box := string(number) + " seen in box " + string(i/3) + "-" + string(j/3)
                
                if seen[row] || seen[col] || seen[box] {
                    return false
                }
                
                seen[row] = true
                seen[col] = true
                seen[box] = true
            }
        }
    }
    
    return true
}

File: 0036-valid-sudoku/0036-valid-sudoku.js
/**
 * @param {character[][]} board
 * @return {boolean}
 */
var isValidSudoku = function(board) {
    let seen = new Set();
    
    for (let i = 0; i < 9; i++) {
        for (let j = 0; j < 9; j++) {
            let number = board[i][j];
            if (number !== '.') {
                let row = number + " seen in row " + i;
                let col = number + " seen in col " + j;
                let box = number + " seen in box " + Math.floor(i / 3) + "-" + Math.floor(j / 3);
                
                if (seen.has(row) || seen.has(col) || seen.has(box)) {
                    return false;
                }
                
                seen.add(row);
                seen.add(col);
                seen.add(box);
            }
        }
    }
    
    return true;
};
